### PR TITLE
fix: apply correct RTL margins

### LINKED ISSUE #274: RTL support?
I'm not sure if I'm configuring this incorrect of that this library doesn't support RTL content.
The following is my layout html:
```html
<div fxLayout="row" fxLayoutAlign="space-between none">
        <div fxFlex="20">
            <md-slide-toggle [(ngModel)]="advanced"></md-slide-toggle>
        </div>
        <div fxFlex="60">
            <span (click)="advanced = !advanced"><i class="fa icon-gear"></i> {{resources.toggleAdvancedSettings}}</span>
        </div>
        <div fxFlex>
            <div fxLayout="col" fxLayoutAlign="start end">
                <span class="cursor-pointer" (click)="closeSidebar()"><i class="fa fa-lg icon-close"></i></span>
            </div>
        </div>
    </div>
```
What I want to accomplish is close to the following:
image
Only the "X" in the image should be aligned to the end, which in RTL case is left instead of right.
I'm using angular 4 with `[dir]="rtl"` directive so I was hoping this is in sync with angular's RTL support...

### PR DESCRIPTION
**Pull request by atscott**
Fixes #274

### CODE REVIEW

**Comment by CaerusKaru** [comment]
The stance is that it is very much encouraged. angular/cdk is a dependency of this library (which may end up getting merged into Angular Material anyway). Please feel free to use it instead of forcing us to maintain two sets of the same code.

File: src/lib/api/directionality.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * The directionality (LTR / RTL) context for the application (or a subtree of it).
+ * Exposes the current direction and a stream of direction changes.
+ */
+@Injectable()
+export class Directionality {
+  /** The current 'ltr' or 'rtl' value. */
+  readonly value: Direction = 'ltr';
+
+  constructor(@Optional() @Inject(DIR_DOCUMENT) _document?: any) {
```
Comment (CaerusKaru) [inline src/lib/api/directionality.ts]: Is there a reason you're not importing the `DOCUMENT` token from `angular/common`? This is why Travis is failing on prerender, it doesn't have access to the global browser `document` object

**Comment by atscott** [comment]
The @angular/cdk was only in dev dependencies (for the demo, I think). I've moved it to the dependency list and am now using the Directionality injectable from the bidi module.

**Comment by atscott** [comment]
FYI - current broken test is also failing in master

**Comment by CaerusKaru** [comment]
Rebase with master should fix broken test

**Comment by atscott** [comment]
Done, thanks.

**Comment by atscott** [comment]
Friendly post-holiday ping. Anything that needs to be done to get this merged? RTL support is pretty important to my team.

**Comment by CaerusKaru** [comment]
It would be great if you could add a demo that displays LTR in the demo-app, just something simple that documents the capability.

Also, please change your commit message to something in line with the Angular contribution guidelines, found [here](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines)

Finally, the PR lists this as a partial fix for #274, are there cases that this does not cover, and can you list that in the commit message?

Other than that the PR LGTM, I'll see what I can do to get it merged in the next week or so (it has to go through internal presubmit tests first)

**Comment by atscott** [comment]
Updated commit message (I think). 

The "partial fix" in the description was from when I wasn't using the angular cdk dir directive. This should be a full fix of what was requested in that bug.

I haven't looked much yet at where to put this in the demo app - I wasn't able to get it to run correctly.

**Comment by CaerusKaru** [comment]
You need to squash your commits into one with a rebase. If there is an issue with the demo-app, can you post it as a new issue so we can take a look?

**Comment by atscott** [comment]
Rebased and filed #547

**Comment by CaerusKaru** [comment]
#547 has been patched in #549. Please incorporate the change (it's one line) and add a demo to the demo-app, leaving out the one-line change in your final commit.

**Comment by atscott** [comment]
Added the demo and updated the implementations to subscribe to the direction changes.

File: src/lib/api/flexbox/layout-gap.ts [hunk 2]
```diff
@@ -67,12 +69,16 @@ export class LayoutGapDirective extends BaseFxDirective implements AfterContentI
...
+              private _zone: NgZone,
+              private readonly _directionality: Directionality) {
     super(monitor, elRef, renderer);
 
     if (container) {  // Subscribe to layout direction changes
       this._layoutWatcher = container.layout$.subscribe(this._onLayoutChange.bind(this));
     }
+
+    this._directionWatcher = 
+        this._directionality.change.subscribe(() => this._updateWithValue());
```
Comment (CaerusKaru) [inline src/lib/api/flexbox/layout-gap.ts]: This should be the same as the _layoutWatcher subscription: 

```typescript
this._directionality.change.subscribe(this._updateWithValue.bind(this));
```

**Comment by CaerusKaru** [comment]
Demo looks great, thanks 👍

**Comment by CaerusKaru** [comment]
@atscott Last thing: lint is failing because of trailing whitespace. Just remove and recommit

**Comment by atscott** [comment]
Thanks for the review and guidance!

**Comment by ThomasBurleson** [comment]
@atscott - this is good work. Thank you.
Currently Flex-Layout can be used standalone. Your additions will add an explicit dependency on the Angular CDK. I am reluctant to add that dependency for only the Directionality.

**Comment by CaerusKaru** [comment]
@ThomasBurleson The PR originally duplicated the CDK directionality code, would you want to incorporate that or a similar patch instead of adding the direct dependency?

IMO adding the CDK as a dependency seems reasonable, especially since a lot of people using Flex Layout likely are already using the CDK with Material (having started with layout using AngularJS Material), but I understand your reservations about adding more dependencies than necessary.

**Comment by CaerusKaru** [comment]
@atscott After discussion, we're going to postpone this at least until the next major release. We want to make sure Flex Layout is more stable before we include the CDK

**Comment by ThomasBurleson** [comment]
@atscott - The reason we want to clone the Directionality class is to avoid introducing an explicit dependency on the CDK at this time. 

> Later, major portions of the library [ e.g.  `/src/lib/mediaquery`] may be migrated to the CDK and then the @angular/flex-layout dependency on the CDK will be acceptable.

**Comment by atscott** [comment]
Alright, that sounds good. I made those changes (had trouble rebasing so kind of redid the whole thing) and I think everything's all set.

**Pull request merged**